crates/security/src/lib.rs

```diff
@@ -37,42 +36,55 @@ use sha2::{Digest, Sha256};
 use sp_core::U256;
 use sp_std::{collections::btree_set::BTreeSet, prelude::*};
 
-/// ## Configuration
-/// The pallet's configuration trait.
-pub trait Config: frame_system::Config {
-    /// The overarching event type.
-    type Event: From<Event> + Into<<Self as frame_system::Config>::Event>;
-}
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
 
-// This pallet's storage items.
-decl_storage! {
-    trait Store for Module<T: Config> as SecurityPallet {
-        /// Integer/Enum defining the current state of the BTC-Parachain.
-        ParachainStatus get(fn status): StatusCode;
-
-        /// Set of ErrorCodes, indicating the reason for an "Error" ParachainStatus.
-        Errors get(fn errors): BTreeSet<ErrorCode>;
-
-        /// Integer increment-only counter, used to prevent collisions when generating identifiers
-        /// for e.g. issue, redeem or replace requests (for OP_RETURN field in Bitcoin).
-        Nonce: U256;
-
-        /// Like frame_system::block_number, but this one only increments if the parachain status is RUNNING.
-        /// This variable is used to keep track of durations, such as the issue/redeem/replace expiry. If the
-        /// parachain is not RUNNING, no payment proofs can be submitted, and it wouldn't be fair to punish
-        /// the user/vault. By using this variable we ensure that they have sufficient time to submit their
-        /// proof.
-        ActiveBlockCount get(fn active_block_number): T::BlockNumber;
+    /// ## Configuration
+    /// The pallet's configuration trait.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
     }
-}
 
-// The pallet's dispatchable functions.
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        type Error = Error<T>;
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    #[pallet::metadata(T::AccountId = "StatusCode")]
```

Comment:
What does this line do?

Reply:
This was an oversight, but for reference > https://crates.parity.io/frame_support/attr.pallet.html#event-palletevent-optional